gtk4/src/custom_layout.rs

```diff
@@ -0,0 +1,73 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use crate::{CustomLayout, Orientation, SizeRequestMode, Widget};
+use glib::translate::*;
+
+impl CustomLayout {
+    #[doc(alias = "gtk_custom_layout_new")]
+    pub fn new<F, G, H>(request_mode: F, measure: G, allocate: H) -> CustomLayout
+    where
+        F: Fn(&Widget) -> SizeRequestMode,
+        G: Fn(&Widget, Orientation, i32, i32, i32, i32, i32),
+        H: Fn(&Widget, i32, i32, i32),
+    {
+        assert_initialized_main_thread!();
+        unsafe {
+            from_glib_full(ffi::gtk_custom_layout_new(
+                Some(request_mode_trampoline::<F>),
+                Some(measure_trampoline::<G>),
+                Some(allocate_trampoline::<H>),
+            ))
+        }
+    }
+}
+
+unsafe extern "C" fn request_mode_trampoline<F>(
+    widget: *mut ffi::GtkWidget,
+    f: glib::ffi::gpointer,
```

Comment:
All these functions don't take a `user_data` or similar, so you can only really pass `fn()` pointers into them so binding this is rather useless as is. Looks like a bad oversight in the C API, or maybe bindings are instead supposed to properly subclass `gtk::LayoutManager` directly. I guess the latter